src/server/database/votes.ts

```diff
@@ -112,33 +119,78 @@ export default class Votes {
             const { insertId } = row;
 
             // If clip has enough votes or is from super user update it
-            this.updateClipStatus(clipId, isSuper, vote);
+            if (updateClip) {
+                this.updateClipStatus(clipId, isSuper, vote);
+            }
 
             return Promise.resolve(insertId);
         } catch (error) {
             return Promise.reject(error);
         }
     };
 
-    addVoteBatch = async (votes: Vote[]): Promise<number> => {
+    addVoteBatch = async (votes: Vote[]): Promise<BatchUploadResults> => {
         try {
+            const output: BatchUploadResults = new BatchUploadResults();
             // Find Marosijo user_client or create a new one if none exists
             const marosijo_client_id = await this.findMarosijoClientId();
+            const validatedIds = await this.getValidatedClips(votes);
+
+            // Store for output
+            output.foundValidatedClips = validatedIds.length;
+
+            // Filter out clips that have been validated
+            const toAdd = votes.filter((vote) => {
+                return !validatedIds.some(
+                    (validated) => validated == vote.clipId
+                );
+            });
+
+            // Create an array for the VALUES that are to be added to the votes table
+            const values: string[] = Array<string>();
+            toAdd.forEach((vote) => {
+                const unsure = ClipVote.UNSURE == vote.vote;
+                const isValid = unsure
+                    ? null
+                    : vote.vote == ClipVote.VALID
+                    ? true
+                    : false;
+                values.push(
+                    `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`
+                );
+            });
+
+            // Add votes to votes table
+            const res = await this.sql.query(`
+                    INSERT INTO
+                        votes (id, clip_id, client_id, is_valid, is_super, is_unsure)
+                    VALUES
+                        ${values.join(',')}
+                    ON DUPLICATE KEY UPDATE
+                        is_valid = VALUES(is_valid),
+                        is_unsure = VALUES(is_unsure)
+
+                        `);
+            output.duplicateVotes = this.getDuplicates(res[0]);
+            output.votesAdded = this.getNewRows(res[0]);
+
+            // Update clips with super votes
+            const supers = toAdd.filter((vote) => vote.isSuper);
```

Comment:
Were the clips which were **not** super excluded from the addition of ordinary votes above?

Reply:
I think I got this now, please read the comment below.

In line 163-173 you update the vote table, in lines 178-186 you are updating the is_valid value of the clips table according to the newly added votes to the vote table? Am I right?

Reply:
Here we have already added the votes to the votes table. Now we need to figure out which votes should cause changes in the clips table. For super votes it is easy since they will determine the is_valid field on their own.

So I handle them first.